src/main/java/com/cx/restclient/ast/AstClient.java

```diff
@@ -268,12 +268,19 @@ private void uploadArchive(byte[] source, String uploadUrl) throws IOException {
         log.info("Uploading the zipped data.");
 
         HttpEntity request = new ByteArrayEntity(source);
+        String baseAstUri = httpClient.getRootUri();
+        httpClient.setRootUri(uploadUrl);
 
-        CxHttpClient uploader = createHttpClient(uploadUrl);
-
-        // Relative path is empty, because we use the whole upload URL as the base URL for the HTTP client.
-        // Content type is empty, because the server at uploadUrl throws an error if Content-Type is non-empty.
-        uploader.putRequest("", "", request, JsonNode.class, HttpStatus.SC_OK, "upload ZIP file");
+        try {
+            // Relative path is empty, because we use the whole upload URL as the base URL for the HTTP client.
+            // Content type is empty, because the server at uploadUrl throws an error if Content-Type is non-empty.
+            httpClient.putRequest("", "", request, JsonNode.class, HttpStatus.SC_OK, "upload ZIP file");
+        }
+        finally {
+            httpClient.setRootUri(baseAstUri);
+        }
```

Comment:
The state of an already existing object is temporarily modified here. Is this fix thread safe?

Also can you please elaborate why we need to reuse the same HTTP client instance here? (I'd expect this to appear in the PR description).